src/MatrixTypes_Native.f90

```diff
@@ -801,5 +815,53 @@ SUBROUTINE transpose_SparseMatrixType(matrix)
       DEALLOCATE(n_Row)
       DEALLOCATE(Row)
     ENDSUBROUTINE transpose_SparseMatrixType
-    
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_SparseMatrixType(matrix)
+    CHARACTER(LEN=*),PARAMETER :: myName='zeroentries_SparseMatrixType'
+      CLASS(SparseMatrixType),INTENT(INOUT) :: matrix
+      REQUIRE(matrix%isInit)
+      matrix%a=0.0_SRK
+    ENDSUBROUTINE zeroentries_SparseMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_DenseSquareMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='zeroentries_DenseSquareMatrixType'
+      CLASS(DenseSquareMatrixType),INTENT(INOUT) :: matrix
+      REQUIRE(matrix%isInit)
+      matrix%a=0.0_SRK
+    ENDSUBROUTINE zeroentries_DenseSquareMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_TriDiagMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='zeroentries_TriDiagMatrixType'
+      CLASS(TriDiagMatrixType),INTENT(INOUT) :: matrix
+      REQUIRE(matrix%isInit)
+      matrix%a=0.0_SRK
+    ENDSUBROUTINE zeroentries_TriDiagMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_DenseRectMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='zeroentries_DenseRectMatrixType'
+      CLASS(DenseRectMatrixType),INTENT(INOUT) :: matrix
+      REQUIRE(matrix%isInit)
+      matrix%a=0.0_SRK
+    ENDSUBROUTINE zeroentries_DenseRectMatrixType
```

Comment:
Question for @stimpsonsg and @aarograh 

Do you think we should include an `ENSURE(ALL(matrix%a == 0.0_SRK))` at the end of these routines?

It seems like it might be kind of redundant for testing, but at the same time, it defines a more explicit contract. I think having more explicit contracts is generally a better thing. Thoughts?

Reply:
I've been sort of indifferent about that kind of thing for the exact combination of reasons you just stated.  I would say that it's probably better to just put the ENSURE there to be completely explicit about the contract (even though I admittedly haven't put similar ENSURE statements in other places), but I'm open to either.

Reply:
Agreed.  It's also a bit more cumbersome to check it from PETSc since it's not as simple as ENSURE(...%a...).  I'd probably leave it off as redundant.

Reply:
Thanks for the comments guys. Sounds like we're pretty much on the same page.